src/components/RedactorWrapper/index.js

```diff
@@ -0,0 +1,187 @@
+import '../../vendor/redactor/_scss/redactor.scss';
+
+export function getEditorContent(editorId) {
+  if (canUseRedactor()) {
+    // use Redactor
+    return window.$R(`#${editorId}`, 'source.getCode');
+  }
+
+  // get content from targets children
+  const node = document.querySelector(`#${editorId}`);
+  if (!node) return '';
+  const children = Array.from(node.children);
+  let stringifiedHTML = '';
+  if (!children.length) {
+    // we have no children so use text content
+    stringifiedHTML = node.textContent;
+  } else {
+    stringifiedHTML = children.reduce(
+      (accumulated, current) => `${accumulated}${current.outerHTML}`,
+      ``,
+    );
+  }
+  return stringifiedHTML;
+}
+
+const defaultRedactorConfig = {
+  settings: {},
+  lang: {},
+};
+
+type PropsShape = {
+  placeholder?: string,
+  content?: string,
+  onChange?: Function,
+  onEditorFocus?: Function,
+  editorId?: string,
+  inputClass?: string,
+  redactorConfig?: {
+    settings: Object,
+    lang: Object,
+  },
+  testId?: string,
+  legacyBehaviour?: boolean,
+};
+
+class RedactorWrapper extends React.Component<PropsShape> {
+  static defaultProps = {
+    placeholder: '',
+    content: '',
+    onChange: () => null,
+    onEditorFocus: () => null,
+    editorId: 'redactor-wrapper',
+    inputClass: 'redactor-wrapper__input',
+    redactorConfig: { settings: {}, lang: {} },
+    testId: 'input-edit-wysiwyg_editor',
+    legacyBehaviour: false,
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      legacyBehaviour: this.props.legacyBehaviour || !canUseRedactor(),
+      hasContent: this.props.content.length > 0,
+    };
+  }
+
+  componentDidMount() {
+    if (!this.state.legacyBehaviour) {
+      this.settings = {
+        ...defaultRedactorConfig.settings,
+        ...this.props.redactorConfig.settings,
+      };
+      this.lang = {
+        ...defaultRedactorConfig.lang,
+        ...this.props.redactorConfig.lang,
+      };
+      RedactorInit();
+
+      (({ onChange, onFocus }) =>
+        $R.add('plugin', 'myplugin', {
+          init(app) {
+            this.app = app;
+          },
+          onchanged(html) {
+            onChange(html);
+          },
+          onblur: () => {
+            onFocus(false);
+          },
+          onfocus: () => {
+            onFocus(true);
+          },
+        }))({ onChange: this.onChangeHandler, onFocus: this.onFocusHandler });
+
+      $R.lang.selected = this.lang;
+      $R(`#${this.props.editorId}`, this.settings);
+    }
+  }
+
+  shouldComponentUpdate(nextProps) {
+    return (
+      this.props.content !== nextProps.content ||
+      this.props.placeholder !== nextProps.placeholder
+    );
+  }
+
+  componentDidUpdate(prevProps) {
+    if (
+      prevProps.placeholder !== this.props.placeholder &&
+      !this.state.hasContent &&
+      !this.state.legacyBehaviour
+    ) {
+      $R(`#${this.props.editorId}`, 'destroy');
+      $R(`#${this.props.editorId}`, this.settings);
```

Comment:
yeah this becomes way more simple if we use `this.props.editorSelector`

Reply:
plz elaborate